test/integration_tests/integrationtest_scanner_api.cpp

```diff
@@ -64,13 +67,41 @@ using namespace ::testing;
 using namespace std::chrono_literals;
 using namespace pilz_testutils;
 
-static monitoring_frame::Message createValidMonitoringFrameMsg()
+static double randDouble(double low, double high)
+{
+  static std::default_random_engine re{};
+  using Dist = std::uniform_real_distribution<double>;
+  static Dist uid{};
+  return uid(re, Dist::param_type{ low, high });
+}
+
+static double restrictToOneDigitsAfterComma(const double& value)
+{
+  return std::round(value * 10.) / 10.;
+}
+
+static std::vector<double> createRandomVec(const unsigned int& num_elements, const double& low, const double& high)
+{
+  std::vector<double> vec(num_elements);
+  // The scanner sends tenth degree values. Therefore, restrict values to one digit after the comma.
```

Comment:
This is correct for the measurements, but not for the intensities. There you need integer values.

Reply:
Done.